pkg/account/deployer/client.go

```diff
@@ -247,6 +249,168 @@ func (d *accountManagementClient) upsertUser(ctx context.Context, userId string)
 	return userId, nil
 }
 
+func (d *accountManagementClient) upsertServiceUser(ctx context.Context, serviceUserId string, data ServiceUser) (remoteId, error) {
+	if serviceUserId == "" {
+		suId, err := d.getServiceUserIDByName(ctx, data.Name)
+		if err != nil {
+			var rnfErr *ResourceNotFoundError
+			if !errors.As(err, &rnfErr) {
+				return "", err
+			}
+
+			return d.createServiceUser(ctx, data)
+		}
+		serviceUserId = suId
+	}
+
+	return d.updateServiceUser(ctx, serviceUserId, data)
+}
+
+func (d *accountManagementClient) createServiceUser(ctx context.Context, dto accountmanagement.ServiceUserDto) (string, error) {
+	uuidDto, resp, err := d.client.ServiceUserManagementAPI.CreateServiceUserForAccount(ctx, d.accountInfo.AccountUUID).ServiceUserDto(dto).Execute()
+	defer closeResponseBody(resp)
+	if err = handleClientResponseError(resp, err, "failed to create service user"); err != nil {
+		return "", err
+	}
+
+	if uuidDto == nil {
+		return "", errors.New("the received data are empty")
```

Comment:
Should the error message point more toward that an ID was not returned from the API?

Reply:
I believe these type of error cannot occur and these checks can be removed. Added to follow ups though